server/routerlicious/packages/routerlicious-base/src/riddler/tenantManager.ts

```diff
@@ -351,6 +422,38 @@ export class TenantManager {
 		return tenantDocument.storage;
 	}
 
+	public async updateKeylessAccessPolicy(
+		tenantId: string,
+		enableKeylessAccess: boolean,
+	): Promise<ITenantConfig> {
+		const latestKeyVersion = this.secretManager.getLatestKeyVersion();
+		let privateKeys: ITenantPrivateKeys | undefined;
+		// If the tenant is being converted to a keyless tenant, generate new private keys, otherwise update them to be undefined
+		if (enableKeylessAccess) {
```

Comment:
Redis and Storage allow us to switch off the policy once it is enabled. However, they do not have an option to "re-enable" the already enabled policy. I can follow the same approach. I will return the existing tenant metadata if the request is to enable the policy for a tenant that already has this policy, or the request is to disable the policy for a tenant and does not have this policy.

I think we can let enableKeylessAccess be false if the policy is disabled. We need not remove it. This is because this prop will eventually be consumed by the portal. Hence, we can let it propagate down.